### ISSUE TITLE
OkHttp causes JVM 11 to hang with HTTP/2

### ISSUE THREAD

**Issue by fastily** [square/okhttp#5832]
The following code
```java
import okhttp3.OkHttpClient;
import okhttp3.Request;

public class A
{
	public static void main(String[] args) throws Throwable
	{
		System.out.println(
				new OkHttpClient.Builder().build().newCall(
						new Request.Builder().url("https://en.wikipedia.org/w/api.php").build()).execute());
	}
}
```
doesn't exit for several minutes in java 11.  I'm using the 4.4.0 release of OkHttp.

Run and compile with `javac A.java && java A`

My environment (macOS Catalina 10.15.3):
```
$ java -version
openjdk version "11.0.2" 2019-01-15
OpenJDK Runtime Environment 18.9 (build 11.0.2+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)

$ echo $BASH_VERSION
5.0.16(1)-release
```

Likely related to #4029, which wasn't actually fixed apparently.

**Comment by swankjesse** [comment]
Try this? https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/#shutdown-isnt-necessary

**Comment by fastily** [comment]
I'm aware that `shutdown()` can be used to exit immediately.  What about the case where I have

```java
// MyLibrary.java
import okhttp3.OkHttpClient;
import okhttp3.Request;

public class MyLibrary
{
	private static final OkHttpClient client = new OkHttpClient.Builder().build();

	public static String getData() throws Throwable
	{
		return client.newCall(new Request.Builder().url("https://en.wikipedia.org/w/api.php").build()).execute().body().string();
	}
}
```
and
```java
// A.java
public class A
{
	public static void main(String[] args) throws Throwable
	{
		System.out.println(MyLibrary.getData());
	}
}
```

If I'm the author of `MyLibrary.java`, and if `A.java` is a program written by another developer consuming `MyLibrary.java`, then requiring a separate, explicit call to `shutdown()` to shutdown OkHttp is effectively a breaking change.

It becomes especially problematic if that other developer has written hundreds of standalone programs with  `MyLibrary.java` as a dependency.  Basically it ensures they will never upgrade.

I'd also like to point out that using OkHttp with HTTP/1.1 doesn't require `shutdown()` whereas HTTP/2 does.  In terms of consistency, this is fairly unintuitive.

**Comment by yschimke** [comment]
PR to use a daemon thread for clients to be consistent for HTTP/2 and HTTP/1.1
https://github.com/square/okhttp/pull/5834

**Comment by swankjesse** [comment]
Fixed in 4.5.1-RC1.
